src/vdaf/poplar1.rs

```diff
@@ -776,69 +775,68 @@ impl Poplar1AggregationParam {
 
 impl Encode for Poplar1AggregationParam {
     fn encode(&self, bytes: &mut Vec<u8>) -> Result<(), CodecError> {
-        // Okay to unwrap because `try_from_prefixes()` checks this conversion succeeds.
-        let prefix_count = u32::try_from(self.prefixes.len()).unwrap();
+        // draft-irtf-cfrg-vdaf-13, Section 8.2.6.6:
+        //
+        // struct {
+        //     uint16_t level;
+        //     uint32_t num_prefixes;
+        //     opaque encoded_prefixes[prefixes_len];
+        // } Poplar1AggParam;
+        //
+        // Level
         self.level.encode(bytes)?;
-        prefix_count.encode(bytes)?;
-
-        // The encoding of the prefixes is defined by treating the IDPF indices as integers,
-        // shifting and ORing them together, and encoding the resulting arbitrary precision integer
-        // in big endian byte order. Thus, the first prefix will appear in the last encoded byte,
-        // aligned to its least significant bit. The last prefix will appear in the first encoded
-        // byte, not necessarily aligned to a byte boundary. If the highest bits in the first byte
-        // are unused, they will be set to zero.
-
-        // When an IDPF index is treated as an integer, the first bit is the integer's most
-        // significant bit, and bits are subsequently processed in order of decreasing significance.
-        // Thus, setting aside the order of bytes, bits within each byte are ordered with the
-        // [`Msb0`](bitvec::prelude::Msb0) convention, not [`Lsb0`](bitvec::prelude::Msb0). Yet,
-        // the entire integer is aligned to the least significant bit of the last byte, so we
-        // could not use `Msb0` directly without padding adjustments. Instead, we use `Lsb0`
-        // throughout and reverse the bit order of each prefix.
-
-        let mut packed = self
-            .prefixes
-            .iter()
-            .flat_map(|input| input.iter().rev())
-            .collect::<BitVec<u8, Lsb0>>();
-        packed.set_uninitialized(false);
-        let mut packed = packed.into_vec();
-        packed.reverse();
-        bytes.append(&mut packed);
+
+        // Number of prefixes
+        let num_prefixes =
+            u32::try_from(self.prefixes.len()).map_err(|e| CodecError::Other(e.into()))?;
+        num_prefixes.encode(bytes)?;
+
+        // Encoded prefixes
+        for prefix in self.prefixes.iter() {
+            bytes.append(&mut prefix.to_bytes());
```

Comment:
@divergentdave I wonder if it would be useful to have unit tests enforce the bit order within each byte.

Reply:
I think the `round_trip_agg_param` test covers this, but a more focused unit test wouldn't hurt.

Reply:
I agree, let's not bother.